nodes.py

```diff
@@ -185,6 +188,13 @@ def descendants(self, selector, **kwargs):
             )
         ]
 
+    def has_parent(self, element, name, attribute, **kwargs):
+        for parent in element.parents:
+            if parent.name == name:
+                if parent.get(attribute[0], "") == attribute[1]:
+                    return True
+        return False
```

Comment:
Non-blocking code golf: could this be done with [`any`](https://docs.python.org/3.12/library/functions.html#any)?

```suggestion
        return any(parent.get(attribute[0], "") == attribute[1] for parent in element.parents if parent.name == name)
```

Reply:
Seems to work!